### ISSUE TITLE
Fix class reference search and add real life demos

### ISSUE THREAD

**Issue by sneakyfildy** [amcharts/amcharts4#154]
Hello, we've been using v3 (commercial) for more than 5 years to build data analytics application.
It is essential that we want to have improved performance using new shiny (not) version of amcharts.

Unfortunately at this point it seems impossible to event estimate LoE required to perform a migration, you've changed all and every aspect, behavior and event names of properties, methods and configs.
It is possible I guess to start a new project with v4, but migration is not :(

The most painful point is that it is impossible event to search for replacements for legacy logic and legacy behavior.

Question:
Is there a possibility to re-make searching through v4 reference to make it search THROUGH REFERENCE, I mean it must search names of props/methods and their descriptions and give a list of what is applicable. Now it gives a pile of articles (10+ pages, with 5 articles on each - it is unusable)

Compare screens from v3 and v4

[https://imgur.com/a/TZIFFG5](https://imgur.com/a/TZIFFG5)

Also there should be demos of real-life usages of charts frameworks: Some API-based application, which uses time series data retrieved from remote host, which pre-processes the data somehow, which changes/removes/adds new "graphs" (series)

Now there is not a single time series chart demo available.

**Comment by martynasma** [comment]
Whoa, slow down :) This is a pre-release version of software we're talking about here.

We didn't change anything from v3. We did make a completely new charting library with v4. There is absolutely nothing that relates to v3 apart from the name. Not a single line of code. The whole thing was planned, constructed and implemented in such a way as to make it super flexible. Something that our users are able to use in ways that were impossible before, and that would allow us implementing new features, chart types, and whatnot at unprecedented speed.

I understand that in order to make something new, we needed to cut lose the legacy config support, which was over 7 years old by then.

That said, I completely agree that we still have long way to go in terms of performance. We're working in it constantly, so things will most definitely bound to dramatically improve.

About documentation. I agree that search could be more useful. We do have filtering search by categories/tags planned. Right now search results favor tutorials ant articles over class reference. This is by design. If you are searching "bullets", you'd want to get the dedicated tutorial about bullets first, not after 100 various classes in classref that mention bullets.

In terms of documentation itself, it was designed to be a one-stop destination for everything. Everything follows unified structure and style. If you take a look at our tutorials, you will find that they are super thorough, with snippets and live demos for everything. Something was definitely lacking in v3 docs.

If you find some concept difficult to grasp, or something does not make sense, just ping us by emailing contact@amcharts.com or opening a new issue here. Someone from our support team will do their best to help you out.

I hope you find this reasonable. Cheers. Enjoy v4 :)
